### PR TITLE
New delphids Cheat Sheet

### PR DESCRIPTION
**Pull request by doriclazar**
Instant Answer Page: https://duck.co/ia/view/delphi_data_types

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @doriclazar

### CODE REVIEW

**Comment by daxtheduck** [comment]
@doriclazar Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by doriclazar** [comment]
Thank you Dax!
I will stay on the noted languanges in the future contributions.
Regardless of me straying from the mission, I updated the IA page. 
-example search didn't contain "cheat sheet" keywords.

File: share/goodie/cheat_sheets/json/delphids.json [hunk 1]
```diff
@@ -0,0 +1,176 @@
+{  
+   "id": "delphids_cheat_sheet",
+   "name": "Delphi data types",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/delphids.json]: lets make this "Delphi Data Types"

File: share/goodie/cheat_sheets/json/delphids.json [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+   "description":"These keywords are used to declare variables using 'var'. There are lots of predefined types and you can also create your own using 'Type' keyword. This cheat sheet contains basic (primitive) types.",
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/delphids.json]: Can you please shorten the description?

File: share/goodie/cheat_sheets/json/delphids.json [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            "key":"LongWord",
+            "val":"0 to 4,294,967,295."
+         },
+         {  
+            "key":"Cardinal",
+            "val":"0 to 4,294,967,295."
+         },
+         {  
+            "key":"LongInt",
+            "val":"-2,147,483,648 to 2,147,483,647."
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/delphids.json]: Can you please remove `.` from the end of all `val`?

File: share/goodie/cheat_sheets/json/delphids.json [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+         },
+         {  
+            "key":"Extended",
+            "val":"19 significant digits, exponent -4932 to +4932."
+         }
+      ],
+      "Text types":[
+         {  
+            "key":"Definition:",
+            "val":"Like many other languages, Delphi allows you to store letters, words, and sentences in single variables. These can be used to display, to hold user details and so on. A letter is stored in a single character variable type, such as Char, and words and sentences stored in string types, such as String."
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/delphids.json]: Too long.

File: share/goodie/cheat_sheets/json/delphids.json [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      "Objects":[  
+         {  
+            "key":"Definition:",
+            "val":"Records are like arrays in that they hold collections of data. However, records can hold a mixture of data types. Ther are a very powerful and useful feature of Delphi, and one that distinguishes Delphi from many other languages."
+         }
+      ],
+      "Files":[  
+         {  
+            "key":"Definition:",
+            "val":"Delphi provides a number of different file access mechanisms. The oldest is in support of consoles, where the Read, ReadLn, Write and WriteLn routines have a syntax that omits the file name. With no file name, IO (Input and Output) is routed to the console."
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/delphids.json]: Too long.

File: share/goodie/cheat_sheets/json/delphids.json [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      "Pointers":[  
+         {  
+            "key":"Definition:",
+            "val":"The Pointer type provides a general use pointer to any memory based variable. That is, one that is accessed by reference. Objects, AnsiStrings, and arrays are examples of reference based variables. But be warned : untyped pointers are dangerous - it is normally always better to use a specific pointer reference to the data type you are using. Only then can you act up on the pointer, as in the example."
+         }
+      ],
+      "Variants":[  
+         {  
+            "key":"Definition:",
+            "val":"The Variant data type provides a flexible general purpose data type. It can hold anything but structured data and pointers. But use with care - there are penalties in performance, potentials for run time errors and poor code clarity when using Variants. Use VarType in conjunction with VarTypeMask to determine the curent data types a Variant is set to.Variants are useful in very specific circumstances, where data types and their content are determined at run time rather than at compile time."
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/delphids.json]: Too long.

**Comment by pjhampton** [comment]
Closing as this PR looks to have gone stale.

If you are looking for something more substantial to work through that will have a big impact for the users of DuckDuckGo, please check out the [DuckDuckHack Issues Table](https://duckduckhack.com/issues). 🚀  😄

**Pull request closed without merging**